test/common/markdown/markdown-spell.tests.ps1

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+Describe "Verify Markdown Spelling" {
+    BeforeAll {
+        if(!(Get-Command -Name 'mdspell' -ErrorAction SilentlyContinue)) {
+            start-nativeExecution {
+                sudo npm install -g markdown-spellcheck@0.11.0
+            }
+        }
+
+        # Cleanup jobs for reliability
+        get-job | remove-job -force
+    }
+
+    AfterAll {
+        # Cleanup jobs to leave the process the same
+        get-job | remove-job -force
+    }
+
+    $groups = Get-ChildItem -Path "$PSScriptRoot\..\..\..\*.md" -Recurse | Where-Object {$_.DirectoryName -notlike '*node_modules*'} | Group-Object -Property directory
+
+    $jobs = @{}
+    # Start all spelling verification in parallel
+    Foreach($group in $groups)
+    {
+        Write-Verbose -verbose "starting jobs for $($group.Name) ..."
+        $job = Start-ThreadJob {
+            param([object] $group)
+            foreach($file in $group.Group)
+            {
+                $results = mdspell --en-us --ignore-numbers --ignore-acronyms --report $file 2>&1
+                Write-Output ([PSCustomObject]@{
+                    file = $file
+                    results = $results
+                })
+            }
+        } -ArgumentList @($group)
+        $jobs.add($group.name,$job)
+    }
+
+    # Get the results and verify
+    foreach($key in $jobs.keys)
+    {
+        $job = $jobs.$key
+        $results = Receive-Job -Job $job -Wait
+        Remove-job -job $Job
+        foreach($jobResult in $results)
+        {
+            $file = $jobResult.file
+            $result = $jobResult.results
+            Context "Verify spelling in $file" {
+                $failures = $result -like '*spelling errors found in*'
```

Comment:
```suggestion
                $failures = @($result) -like '*spelling errors found in*'
```
If you don't force this as an array and there is only one or zero items in $result, it will return $false, which is one of the symptoms of the error (loop is passing `$false` as a value through to the pester test which is throwing).